test/cpp/jit/test_ir.cpp

```diff
@@ -92,5 +92,88 @@ void testBlocks() {
       ->run(*g2);
 }
 
+void testCommonAncestor() {
+  std::string input_str = R"(
+graph(%x : Tensor,
+      %a.1 : bool,
+      %b.1 : bool,
+      %c.1 : bool):
+  %4 : int = prim::If(%a.1)
+    block0():
+      %5 : int = prim::If(%b.1)
+        block0():
+          %6 : int = prim::Constant[value=2]()
+          -> (%6)
+        block1():
+          %7 : int = prim::Constant[value=3]()
+          -> (%7)
+      -> (%5)
+    block1():
+      %8 : int = prim::If(%c.1)
+        block0():
+          %9 : int = prim::Constant[value=4]()
+          -> (%9)
+        block1():
+          %10 : int = prim::Constant[value=5]()
+          -> (%10)
+      -> (%8)
+  return (%4)
+)";
+
+  torch::jit::Graph g;
+  torch::jit::script::parseIR(input_str, &g);
+
+  std::vector<std::string> value_names{"6", "7", "9", "10"};
```

Comment:
`parseIR` has another API that fills out a map with string, IValue * pairs:
`parseIR(
    const std::string& str,
    torch::jit::Graph* graph,
    std::unordered_map<std::string, Value*>& vmap)` , can you use that instead of `populate_values`